### ISSUE TITLE
python bindings install fails on OS X

### ISSUE THREAD

**Issue by ahundt** [jrl-umi3218/SpaceVecAlg#16]
I'm trying to install the SpaceVecAlg python bindings on OS X, but it seems to pass the `-framework` parameter to clang incorrectly in this case:

```
[100%] Generating local SpaceVecAlg Python bindings
cd /Users/athundt/src/jrl-umi3218/SpaceVecAlg/binding/python && python setup.py build_ext --inplace
/usr/local/lib/python2.7/site-packages/setuptools/dist.py:360: UserWarning: The version specified ('1.0.0-b18e913') is an invalid version, this may not work as expected with newer versions of setuptools, pip, and PyPI. Please see PEP 440 for more details.
  "details." % self.metadata.version
running build_ext
building 'sva.sva' extension
clang -fno-strict-aliasing -fno-common -dynamic -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/athundt/src/jrl-umi3218/SpaceVecAlg/build -I/Users/athundt/src/jrl-umi3218/SpaceVecAlg/build/include -I/Users/athundt/src/jrl-umi3218/SpaceVecAlg/include -I/usr/local/Cellar/eigen/3.3.4/include/eigen3 -I/Users/athundt/src/jrl-umi3218/SpaceVecAlg/src -I/Users/athundt/source/jrl-umi3218/SpaceVecAlg/binding/python/include -I/usr/local/lib/python2.7/site-packages/numpy/core/include -I/usr/local/include -I/usr/local/opt/openssl/include -I/usr/local/opt/sqlite/include -I/usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c sva/sva.cpp -o build/temp.macosx-10.11-x86_64-2.7/sva/sva.o -std=c++11 -framework
clang: error: argument to '-framework' is missing (expected 1 value)
error: command 'clang' failed with exit status 1
make[2]: *** [binding/python/CMakeFiles/sva-python-bindings] Error 1
make[1]: *** [binding/python/CMakeFiles/sva-python-bindings.dir/all] Error 2
make: *** [all] Error 2

```

I'm using the homebrew installed python:
```
athundt at Andrews-2013-MacBook-Pro-2 in ~/src/jrl-umi3218/SpaceVecAlg/build on master!
± python --version
Python 2.7.13

athundt at Andrews-2013-MacBook-Pro-2 in ~/src/jrl-umi3218/SpaceVecAlg/build on master!
± which python
/usr/local/bin/python
```

Here is the complete install script I use:
https://github.com/ahundt/robotics_setup/blob/master/robotics_tasks.sh

**Comment by ahundt** [comment]
I'm looking into this a bit and the problem might be due to FindPythonLibs.cmake, perhaps this version might work better?
https://github.com/bulletphysics/bullet3/tree/master/build3/cmake

Some things seem to be wrong in tricky ways in the CMake provided version of these scripts.

**Comment by gergondet** [comment]
Hi Andrew,

I will send patch the issue and add MacOS to the CI later on but can you remove lines 65-76 in `binding/python/setup.in.py`. This should do the trick.

**Comment by ahundt** [comment]
@gergondet can you apply this change? It seems to still be outstanding

**Comment by ahundt** [comment]
Based on this I've created some pull requests to take steps towards solving the problems. See:
https://github.com/jrl-umi3218/jrl-cmakemodules/pull/124
https://github.com/jrl-umi3218/SpaceVecAlg/pull/21

Some additional steps will likely be needed to actually integrate the cmake modules changes and use the variables defined in each library.

**Comment by ahundt** [comment]
Also https://github.com/jrl-umi3218/RBDyn/pull/44

**Comment by gergondet** [comment]
Hi Andrew, sorry about that. I'm pushing the patches now :+1:

**Comment by ahundt** [comment]
@gergondet Thanks for the fixes! 3 more to go:  jrl-umi3218/eigen-qld#8,  jrl-umi3218/mc_rbdyn_urdf#5, jrl-umi3218/Tasks#37. :-)
